src/features/groups/list/group-details/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { useTranslation } from 'react-i18next';
+
+import { Row } from '../../../../components/Containers';
+import { ITransactionGroup } from '../../types';
+import {
+  AddMembersWrapper,
+  ContentWrapper,
+  GroupCurrencyText,
+  GroupDetailsBackground,
+  GroupDetailsWrapper,
+  GroupInfoWrapper,
+  MainInfoWrapper,
+  SaveButtonWrapper
+} from './index.styled';
+import addAvatar from '../../../../assets/images/add-avatar.png';
+import arrowRight from '../../../../assets/images/arrow-right.png';
+import Select from '../../../../components/Select';
+import { TextUnderline } from '../../../../components/Text';
+import currencies from '../../../../constants/currencies';
+import { useCreateTransactionGroupMutation, useLazyGetTransactionGroupByIdQuery, useUpdateTransactionGroupMutation } from '../../api';
+import Button from '../../../../components/Button';
+import { requiredValidator } from '../../../auth/validation';
+import Input from '../../../../components/Input';
+import GroupMembers from '../group-members';
+
+interface IValidation {
+  name: boolean;
+}
+
+export interface IGroupDetailsProps {
+  groupId?: string | null;
+  onSaved: (group: ITransactionGroup) => void;
+  isAddGroupMembersMode: boolean;
+  onGroupMembersModeChange: (value: boolean) => void;
+}
+
+const GroupDetails = ({ groupId, onSaved, isAddGroupMembersMode, onGroupMembersModeChange }: IGroupDetailsProps) => {
+  const { t } = useTranslation('groups');
+
+  const [getGroupById] = useLazyGetTransactionGroupByIdQuery();
+  const [createGroup, { data: createdGroup }] = useCreateTransactionGroupMutation();
+  const [updateGroup, { data: updatedGroup }] = useUpdateTransactionGroupMutation();
+
+  const [form, setForm] = useState<ITransactionGroup>({
+    id: '',
+    name: '',
+    currencyCode: currencies[0].value,
+    participants: [],
+    transactions: [],
+    ownerId: ''
+  });
+
+  const [validation, setValidation] = useState<IValidation>({
+    name: false
+  });
+
+  const isDisabled = useMemo(() => Object.values(validation).some(el => !el), [validation]);
+
+  const onValidationChange = useCallback((name: string, value: boolean) => {
+    setValidation(validation => ({
+      ...validation,
+      [name]: value,
+    }));
+  }, []);
+
+  const onFormSubmit = async () => {
+    if (form.id) {
+      updateGroup({
+        name: form.name,
+        currencyCode: form.currencyCode,
+        id: form.id
+      });
+    } else {
+      createGroup({
+        name: form.name,
+        currencyCode: form.currencyCode
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (createdGroup) {
+      onSaved(createdGroup);
+    }
+  }, [createdGroup, onSaved]);
+
+  useEffect(() => {
+    if (updatedGroup) {
+      onSaved(updatedGroup);
+    }
+  }, [updatedGroup, onSaved]);
+
+  useEffect(() => {
+    if (groupId) {
+      getGroupById(groupId).then(({ data }) => {
+        if (data) {
+          setForm(data);
+        }
+      });
```

Comment:
Try don't use data from here. Use data from `useLazyGetTransactionGroupByIdQuery` directly. There you can get data and error separately.
Here if something happens you will get an error in `data` object, and you will try to set it as form...